File: 15-10-2025_Adjacent_Increasing_Subarrays_Detection_II.py
class Solution:
    def maxIncreasingSubarrays(self, nums: List[int]) -> int:
        n = len(nums)
        cnt = 1
        prev = 0
        max_val = 0

        for i in range(1, n):
            if nums[i] > nums[i - 1]:
                cnt += 1
            else:
                prev = cnt
                cnt = 1

            temp = prev
            if cnt < prev:
                temp = cnt

            if temp > max_val:
                max_val = temp

            if cnt // 2 > max_val:
                max_val = cnt // 2

        return max_val

File: 15-10-2025_Adjacent_Increasing_Subarrays_Detection_II.java
class Solution {
    public int maxIncreasingSubarrays(int[] nums) {
        int n = nums.length;
        int cnt = 1, prev = 0;
        int max = 0;

        for (int i = 1; i < n; ++i) {
            if (nums[i] > nums[i - 1]) {
                cnt++;
            } else {
                prev = cnt;
                cnt = 1;
            }

            int temp = prev;
            if (cnt < prev) {
                temp = cnt;
            }

            if (temp > max) {
                max = temp;
            }

            if (cnt / 2 > max) {
                max = cnt / 2;
            }
        }

        return max;
    }
}
